gateway/ln-gateway/src/lib.rs

```diff
@@ -126,18 +102,24 @@ impl Gateway {
             module_gens: module_gens.clone(),
         };
 
-        gw.load_federation_actors(decoders, module_gens, route_hints)
-            .await;
+        gw.load_federation_actors(decoders, module_gens).await?;
 
-        gw
+        Ok(gw)
     }
 
     async fn load_federation_actors(
         &self,
         decoders: ModuleDecoderRegistry,
         module_gens: ClientModuleGenRegistry,
-        route_hints: Vec<RouteHint>,
-    ) {
+    ) -> Result<()> {
+        // Fetch route hints form the LN node
+        let route_hints: Vec<RouteHint> = self
+            .lnrpc
+            .routehints()
+            .await
+            .map_err(|_| GatewayError::FailedToFetchRouteHints)?
+            .try_into()
+            .map_err(|_| GatewayError::FailedToFetchRouteHints)?;
```

Comment:
Why would different GW actors talking to the same LN node, being loaded a few ms after each other request route hints multiple times instead of only once?

Reply:
They don't. This function is called once in `Gateway::new()` -- not per-actor.

The reason I moved if from the constructor to this method was because this method was the only part of `Gateway::new()` that cared about route hints. Seems more self-contained to call it inside this function.

Reply:
I think it might be better if the gateway had the route hints attribute, and the actors were simpler clients without this state.

Reply:
Then we'll need some synchronization primitive again that preferably won't lead to deadlocks in some obscure way :grimacing: But it would make things easier, like updating the available route hints every now and then.

Reply:
> They don't. This function is called once in `Gateway::new()` -- not per-actor.

Also sorry for the confusion, should have read more carefully.